### DISCUSSION TITLE
mpv stopped working today after running pkg upgrade

### DISCUSSION THREAD

**Discussion by kriskropd** [termux/termux-packages#7533] [Q&A] [answered]
To recreate.

```
pkg update
pkg install mpv
```

When trying to run any mpv command, this message appears.

```
> mpv --version
CANNOT LINK EXECUTABLE "mpv": "/data/data/com.termux/files/usr/lib/libass.so" has text relocations (https://android.googlesource.com/platform/bionic/+/master/android-changes-for-ndk-developers.md#Text-Relocations-Enforced-for-API-level-23)
```

```
> pkg show mpv
Package: mpv
Version: 0.33.1
Maintainer: @termux
Installed-Size: 3916 kB
Depends: ffmpeg, libandroid-glob, libandroid-support, libarchive, libcaca, libiconv, liblua52, pulseaudio, openal-soft, zlib
Homepage: https://mpv.io/
Download-Size: 948 kB
APT-Manual-Installed: yes
APT-Sources: https://deb.kcubeterm.me/termux-main stable/main i686 Packages
Description: Command-line media player
```

I do not have scanelf available, so I ran the readelf alternative
```
 > readelf --dynamic /data/data/com.termux/files/usr/lib/libass.so | grep TEXTREL
 0x00000016 (TEXTREL)                    0x0
 0x0000001e (FLAGS)                      TEXTREL BIND_NOW
```

The [topic URL](https://android.googlesource.com/platform/bionic/+/master/android-changes-for-ndk-developers.md#Text-Relocations-Enforced-for-API-level-23) provided in error output recommends a solution.

> Potential problems: Relocations enforce code pages being writable, and wastefully increase the number of dirty pages in memory. The dynamic linker has issued warnings about text relocations since Android K (API 19), but on API 23 and above it refuses to load code with text relocations.
> 
> Resolution: rewrite assembler to be position independent to ensure no text relocations are necessary. [The Gentoo Textrels](https://wiki.gentoo.org/wiki/Hardened/Textrels_Guide) guide has instructions for fixing text relocations, and more detailed [scanelf documentation](https://wiki.gentoo.org/wiki/Hardened/PaX_Utilities).

This thread is the most involved I've been with the mpv package or project. So "rewriting the assembler" and contributing that upstream is a bit out-of-scope. Any suggestions for a more lay-user like me? I don't suspect downgrading is a practical option.

**Comment by tomty89** [comment, marked as answer]
I think you should just file an issue / bug report here. Often case like this can be worked around by disabling assembly.

**Reply by jeevatattva** [reply]
>I think you should just file an issue / bug report here.

I guess, GitHub may have suggested to start a discussion when OP might have wanted to create an issue. Happened to me the first time, when I wanted to create an issue for `libx265 3.5`.

>Often case like this can be worked around by disabling assembly.

Yes, `libass` has asm code only for `x86`.

@kriskropd please create an issue, fix would be simply to disable asm.

**Reply by kriskropd** [reply]
I've created Issue https://github.com/termux/termux-packages/issues/7648 with this suggestion. Thanks for the input.
